packages/shell-api/src/stream-processor.ts

```diff
@@ -57,6 +58,43 @@ export default class StreamProcessor extends ShellApiWithMongoClass {
     });
   }
 
+  /**
+   * modify can be used like so:
+   *  Change the pipeline:
+   *    sp.name.modify(pipeline)
+   *  Change the pipeline with additional options:
+   *    sp.name.modify(pipeline, {resumeFromCheckpoint: false})
+   *  For modify requests that don't change the pipeline:
+   *    sp.name.modify({resumeFromCheckpoint: false})
+   */
+  @returnsPromise
+  async modify(
+    pipelineOrOptions: Document[] | Document,
+    options: Document = {}
+  ) {
+    if (Array.isArray(pipelineOrOptions)) {
+      options['pipeline'] = pipelineOrOptions;
+    } else if (typeof pipelineOrOptions === 'object') {
+      if (Object.keys(options).length !== 0) {
```

Comment:
```suggestion
      if (options) {
```

Reply:
I think `if (options) {}` evaluates to true when options is `{}` (the default argument). So, if I make this change,  I'll need to change `options: Document = {}` to `options: Document = null`. Do you have a preference? I think keeping `options: Document = {}` and `if (Object.keys(options).length !== 0) ` is slightly better.

Apologies if I'm missing something-- this is the first typescript PR I've written :)

Reply:
Yeah, the `options: Document = {}` above would become `options?: Document` in this case 



> I think keeping `options: Document = {}` and `if (Object.keys(options).length !== 0) ` is slightly better.

I'd say checking the # of keys of an argument to check whether it was passed or not is definitely a bit unidiomatic. For example, it will just throw an unexpected exception if somebody passes `null` as an option objects.

Reply:
Ah yes that is much better, thanks. I didn't realize that options?: Document would work with the spread operator in `{...options, pipeline: pipelineOrOptions }`.